index.ts

```diff
@@ -91,6 +116,55 @@ export const sandbox = {
         )
       )
     )
+
+    if (tableStreams) {
+      // Reset Stream defaults
+      for (const arcStream of tableStreams) {
+        shardMap[arcStream.table] = []
+        await resetTableStreams(
+          dynamodbClient,
+          ddbStreamsClient,
+          arcStream.table
+        )
+      }
+
+      // eslint-disable-next-line no-constant-condition
+      while (true) {
```

Comment:
Because this is an infinite loop, `start()` will never return. I think that will break Architect's functionality of automatically starting and stopping the sandbox on certain file changes.

Instead, you could have a timeout handler that recursively calls setTimeout so that your table streams code runs in the background, as it were.

You need the `end()` entry point to stop the table streams invocation code. It could do that by setting a sentinel variable that the table streams invocation loop checks periodically.

Reply:
I've not noticed any issues with this in testing. The other version of the stream plugin used callbacks and it was often getting messed up and out of sync with how it would attempt to call the stream functions

Reply:
Did you test making changes to files that would have caused Architect to restart the sandbox plugins? Modifying app.arc and changing the `@tables` or `@tables-streams` pragmas would do that. I see no way that it would work correctly.

Reply:
I see, working on a change